segway/input_master.py

```diff
@@ -209,131 +181,303 @@ def get_template_component_suffix(self, component_number):
+                        names.append("mean_seg{}_subseg{}_{}{}".format(i, j, 
+                                                                   track_name, 
+                                                                  component_suffix))
+
+        return names
+      
+    def generate_mx_object_names(self):
+        """
+        Generate MX object names. 
+        """      
+        head_track_names = self.get_head_track_names()
+        names = []
+        for i in range(self.num_segs):
+            for j in range(self.num_subsegs):
+                for track_name in head_track_names:
+                    names.append("mx_seg{}_subseg{}_{}".format(i, j, track_name))
+
+        return names
+    
+    def generate_diag_gaussian_mc_object_names(self):
+        """
+        Generate DiagGaussianMC object names. 
+        """      
+        head_track_names = self.get_head_track_names()
+        names = []
+        for component_number in range(self.num_mix_components):
+            for i in range(self.num_segs):
+                for j in range(self.num_subsegs):
+                    for track_name in head_track_names:
+                        component_suffix = self.get_template_component_suffix(component_number)
+                        names.append("mc_{}_seg{}_subseg{}_{}{}".format(self.distribution, 
+                                                                    i, j, track_name, 
+                                                                   component_suffix))
+        return names
+    
+    def generate_gamma_mc_object_names(self):
+        # todo: gammascale, gammashape
+        pass
+    
+    def generate_missing_mc_object_names(self):
+        pass
 
-    def make_segCountDown_tree_spec(self, resourcename):  # noqa
+    def generate_dpmf_object_names(self):
+        """
+        Generate DPMF object names. 
+        """      
+        names = []
+        if self.num_mix_components == 1:
+            names.append("dpmf_always")
+        else: 
+            names.append("")
+            # TODO (with dirichlet extra rows) 
+        
+        return names 
+      
+    def generate_gmtk_object_names(self, gmtk_object_type):
+        """
+        Generate GMTK object names for the types:
+        NameCollection: "col"
+        entries in NameCollection: "mx_name"
+        Covar: "covar", "tied_covar"
+        Mean: "mean"
+        MX: "mx"
+        MC: "diag_gaussian_mc", "gamma_mc", "missing_mc", "gammascale"
+        DPMF: "dpmf"
+        :param gmtk_object_type: str: type of gmtk object for which names must be generated
+        :return: list[str]: list of GMTK object names 
+        """
+        allowed_types = ["mx", "diag_gaussian_mc", "gamma_mc", "missing_mc", "mean",
+                     "covar", "collection_names", "collection_entries", "dpmf", "gammascale",
+                     "gammashape", "tied_covar"]
+        
+        if not gmtk_object_type in allowed_types:
```

Comment:
`allowed_types` is not necessary here. You can check if the key exists in `GMTK_OBJECT_NAME_GENERATORS`